eng/native/init-os-and-arch.sh

```diff
@@ -35,6 +35,10 @@ fi
 case "$CPUName" in
```

Comment:
it would be nice to remove this script now that we have it in eng/common

Reply:
I will check if we can remove the extra copy.

Similarly, any thoughts about the `dotnet-install.sh` copy? It’s not used at the moment.

Reply:
`eng/common/dotnet-install.sh` is used by `eng/common/tools.sh`.

Reply:
It's not used by the top-level `dotnet.sh` which bit me quite a bit.

Reply:
On second look, the top-level `dotnet.sh` references `eng/common/tools.sh` which in turn downloads the official `dotnet-install.sh` from web.

Reply:
It's conditional on `-a` (if path exists) in tools.sh, so not unused. Maybe it is there for SB offline mode or some other repo use-case (like https://github.com/dotnet/arcade/issues/14283), so leave it for now as out of PR scope? init-os-and-arch is definitely something we can dedup (as it was the plan https://github.com/dotnet/arcade/pull/13947). It can also be cleaned up as a separate PR.

Reply:
> It's conditional on -a (if path exists) in tools.sh, so not unused. 

The `$root` path it checks is outside of `eng/common` normally though (e.g. in runtime we install the local dotnet into `<repo root>/.dotnet`) so the copy in common won't be used by tools.sh.

It is used by the Arcade msbuild targets to install dotnet: https://github.com/dotnet/arcade/blob/2fb543a45580400a559b5ae41c96a815ea14dac5/src/Microsoft.DotNet.Arcade.Sdk/tools/InstallDotNetCore.targets#L6-L11

Reply:
I guess that begs a question... shouldn't it be used? (not going to resolve it in this PR, just asking)

Reply:
No idea. From a quick look the eng/common/dotnet-install.sh is _much_ simpler than the one from https://dot.net/v1/dotnet-install.sh, that was probably intentional?